src/rust/util/src/cell.rs

```diff
@@ -12,48 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Same as `core::sync::SyncUnsafeCell`, which is still in nightly. Can remove once it is stable.
-pub struct SyncUnsafeCell<T: ?Sized> {
+// Link library to provide symbols necessary for critical section even if code isn't called
+// explicitly. See
+// https://github.com/rust-embedded/critical-section/blob/bc0d1736c3b14e340c1b0c29042edb3e6bffe6b8/README.md#undefined-reference-errors
+#[cfg(not(any(feature = "testing", feature = "c-unit-testing")))]
+#[allow(unused_imports)]
+use cortex_m as _;
+
+// A cell useful for global mutable state.
+pub struct SyncCell<T: ?Sized> {
     value: core::cell::UnsafeCell<T>,
 }
 
 // Implement Sync if the wrapped type is Sync.
-unsafe impl<T: ?Sized + Sync> Sync for SyncUnsafeCell<T> {}
+unsafe impl<T: ?Sized + Sync> Sync for SyncCell<T> {}
 
-impl<T> SyncUnsafeCell<T> {
+impl<T> SyncCell<T> {
     pub const fn new(val: T) -> Self {
-        SyncUnsafeCell {
+        SyncCell {
             value: core::cell::UnsafeCell::new(val),
         }
     }
 
     /// Reads the value from `self` without moving it. This leaves the
     /// memory in `self` unchanged.
```

Comment:
Should we maybe demand that `T: Copy`? Making a clone of something that can be dropped seems scary?